src/ls_server.rs

```diff
@@ -251,34 +252,36 @@ impl LsService {
     }
 
     fn init(&self, id: usize, init: InitializeParams) {
-        let result = InitializeCapabilities {
+        let result = InitializeResult {
             capabilities: ServerCapabilities {
-                textDocumentSync: DocumentSyncKind::Incremental as usize,
-                hoverProvider: true,
-                completionProvider: CompletionOptions {
-                    resolveProvider: true,
-                    triggerCharacters: vec![".".to_string()],
-                },
+                text_document_sync: Some(TextDocumentSyncKind::Incremental),
```

Comment:
I mean instead of having

```
enum Foo {
    Bar,
    Baz,
}
```

and using `Option<Foo>`, you have:

```
enum Foo {
    Bar,
    Baz,
    None, // or Null or Missing or whatever
}
```

and use `Foo` directly.

> smells like a null pointer.

There is nothing wrong with null pointers, the problem is when they are implicit and thus un-tested. With this pattern, they are just as explicit and thus safe to use.